src/main/scala/org/virtuslab/yaml/internal/load/parse/ParserImpl.scala

```diff
@@ -33,14 +34,15 @@ private enum Production:
  * ParseDocumentStart    ::= <ParseNode> ParseDocumentEnd
  * ParseDocumentStartOpt ::= epsilon | <ParseDocumentStart> <ParseDocumentStartOpt>
  * ParseNode             ::= <ParseMappingStart> | <ParseFlowMappingStart> | <ParseSequenceStart> | ParseScalar
+ * ParseNodeOpt          ::= epsilon | <ParseMappingStart> | <ParseFlowMappingStart> | <ParseSequenceStart> | ParseScalar
  * ParseMappingStart     ::= <ParseKey> ParseMappingEnd
  * ParseFlowMappingStart ::= <ParseFlowMappingEntry> ParseFlowMappingEnd
  * ParseFlowMappingEntry ::= <ParseScalar> | <ParseFlowMappingEntry> <ParseFlowMappingEnd> <ParseNode>
- * ParseSequenceStart    ::= <ParseNode> <ParseSequenceEntryOpt> ParseSequenceEnd
+ * ParseSequenceStart    ::= <ParseNodeOpt> <ParseSequenceEntryOpt> ParseSequenceEnd
```

Comment:
Grammar is incorrect because `FlowSequence` can be empty but `Block Sequence` can't. These two kinds of sequences should be distiguished.

Reply:
Right, it was mistake, so I added new production state `ParseFlowSequenceStart` which allows to empty sequences.